src/dagmc/DagMC.cpp

```diff
@@ -212,6 +214,423 @@ ErrorCode DagMC::setup_indices() {
+    - Construct the new volume's OBBTree/BVH
+    - Construct the updated implicit complement OBBTree/BVH
+  */
+  ErrorCode rval;
+
+  // remove existing graveyard
+  if (overwrite) {
+    remove_graveyard();
+  }
+
+  // if a graveyard already exists and we aren't overwriting it,
+  // report an error
+  if (has_graveyard() && !overwrite) {
+    MB_CHK_SET_ERR(MB_FAILURE, "Graveyard already exists");
+  }
+
+  // currently relying on the BVH as looping over all vertices may
+  // be prohibitive
+  if (!geom_tool()->have_obb_tree()) {
+    MB_CHK_SET_ERR(MB_FAILURE, "Graveyard creation attempted without BVH");
+  }
+
+  BBOX box;
+  for (int i = 0; i < num_entities(3); i++) {
+    // get the bounding box of the volume
+    moab::EntityHandle vol = this->entity_by_index(3, i + 1);
+    double vmin[3], vmax[3];
+    rval = this->getobb(vol, vmin, vmax);  // this method name is a misnomer
+    MB_CHK_SET_ERR(rval, "Failed to get volume bounding box");
+    // update the global bounding box
+    box.update(vmin);
+    box.update(vmax);
+  }
+
+  if (!box.valid()) {
+    MB_CHK_SET_ERR(rval,
+                   "Invalid model bounding box generated for graveyard volume");
+  }
+
+  // create a new volume meshset
+  EntityHandle volume_set;
+  rval = MBI->create_meshset(0, volume_set);
+  MB_CHK_SET_ERR(rval, "Failed to create a graveyard volume set");
+
+  // add volume set to the model
+  rval = geom_tool()->add_geo_set(volume_set, 3);
+  MB_CHK_SET_ERR(rval, "Failed to add the volume to the GeomTopoTool");
+
+  // set the category tag
+  std::string volume_str;
+  volume_str.resize(CATEGORY_TAG_SIZE);
+  volume_str = "Volume";
+  rval = MBI->tag_set_data(category_tag(), &volume_set, 1, volume_str.c_str());
+  MB_CHK_SET_ERR(rval, "Failed to set graveyard volume category");
+
+  // create group set for the graveyard volume
+  EntityHandle group_set;
+  rval = MBI->create_meshset(0, group_set);
+  MB_CHK_SET_ERR(rval, "Failed to create a new graveyard group set");
+
+  rval = geom_tool()->add_geo_set(group_set, 4);
+  MB_CHK_SET_ERR(rval, "Failed to add the graveyard group to the GeomTopoTool");
+
+  // set the group category
+  std::string group_str;
+  group_str.resize(CATEGORY_TAG_SIZE);
+  group_str = "Group";
+  rval = MBI->tag_set_data(category_tag(), &group_set, 1, group_str.c_str());
+  MB_CHK_SET_ERR(rval, "Failed to set the group category");
+
+  // set the volume name tag data (material metadata)
+  rval = MBI->tag_set_data(name_tag(), &group_set, 1, GRAVEYARD_NAME.c_str());
+  MB_CHK_SET_ERR(rval, "Failed to set the graveyard name");
+
+  // add the graveyard volume to this group
+  rval = MBI->add_entities(group_set, &volume_set, 1);
+  MB_CHK_SET_ERR(rval,
+                 "Failed to add the graveyard volume to the graveyard group");
+
+  /// SURFACE CREATION ///
+
+  // expand the box a bit
+  for (int i = 0; i < 3; i++) {
+    double bump = 10 * numerical_precision();
+    box.upper[i] += bump;
+    box.lower[i] -= bump;
+  }
```

Comment:
maybe the 2 loop to increase the size of of the box could be part of a dedicated method ?

Reply:
mayb as part of the `BOX` struc ?

Reply:
- [ ] Good suggestion. Better to put something like that in the struct for use elsewhere.